frame/vesting/src/weights.rs

```diff
@@ -44,103 +44,157 @@ use sp_std::marker::PhantomData;
-		(45_324_000 as Weight)
-			// Standard Error: 12_000
-			.saturating_add((214_000 as Weight).saturating_mul(l as Weight))
+	fn vest_other_unlocked(l: u32, s: u32, ) -> Weight {
+		(59_567_000 as Weight)
+			// Standard Error: 2_000
+			.saturating_add((148_000 as Weight).saturating_mul(l as Weight))
+			// Standard Error: 6_000
+			.saturating_add((84_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(3 as Weight))
 			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 	}
-	fn vested_transfer(l: u32, ) -> Weight {
-		(96_661_000 as Weight)
-			// Standard Error: 10_000
-			.saturating_add((211_000 as Weight).saturating_mul(l as Weight))
+	fn vested_transfer(l: u32, _s: u32, ) -> Weight {
+		(105_528_000 as Weight)
+			// Standard Error: 6_000
+			.saturating_add((149_000 as Weight).saturating_mul(l as Weight))
 			.saturating_add(T::DbWeight::get().reads(3 as Weight))
 			.saturating_add(T::DbWeight::get().writes(3 as Weight))
 	}
-	fn force_vested_transfer(l: u32, ) -> Weight {
-		(98_812_000 as Weight)
-			// Standard Error: 13_000
-			.saturating_add((139_000 as Weight).saturating_mul(l as Weight))
+	fn force_vested_transfer(l: u32, _s: u32, ) -> Weight {
+		(106_006_000 as Weight)
+			// Standard Error: 9_000
+			.saturating_add((111_000 as Weight).saturating_mul(l as Weight))
 			.saturating_add(T::DbWeight::get().reads(4 as Weight))
 			.saturating_add(T::DbWeight::get().writes(4 as Weight))
 	}
+	fn not_unlocking_merge_schedules(l: u32, s: u32, ) -> Weight {
+		(73_233_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((147_000 as Weight).saturating_mul(l as Weight))
+			// Standard Error: 5_000
+			.saturating_add((143_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	}
+	fn unlocking_merge_schedules(l: u32, s: u32, ) -> Weight {
+		(73_772_000 as Weight)
+			// Standard Error: 2_000
+			.saturating_add((145_000 as Weight).saturating_mul(l as Weight))
+			// Standard Error: 6_000
+			.saturating_add((119_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(T::DbWeight::get().reads(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	}
 }
 
 // For backwards compatibility and tests
 impl WeightInfo for () {
-	fn vest_locked(l: u32, ) -> Weight {
-		(42_905_000 as Weight)
-			// Standard Error: 13_000
-			.saturating_add((232_000 as Weight).saturating_mul(l as Weight))
+	fn vest_locked(l: u32, s: u32, ) -> Weight {
+		(60_524_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((158_000 as Weight).saturating_mul(l as Weight))
+			// Standard Error: 5_000
+			.saturating_add((181_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(RocksDbWeight::get().reads(2 as Weight))
-			.saturating_add(RocksDbWeight::get().writes(1 as Weight))
+			.saturating_add(RocksDbWeight::get().writes(2 as Weight))
```

Comment:
is this additional write accounted for?

Reply:
Yes - we used to read in the schedules and the lock, but then only update the lock. We now update the schedules (delete ones that have ended) in addition to updating the lock.